### DISCUSSION TITLE
Jinja blocks should be indented

### DISCUSSION THREAD

**Discussion by tconbeer** [tconbeer/sqlfmt#317] [The sqlfmt Style] [answered]
We calculate the (nesting) depth of jinja blocks, but we don't use this for indentation.

I think this makes the most sense for simple use cases when you're actually templating SQL, since it makes the structure of the SQL more apparent, and the compiled code will be indented properly:

```sql
{% set cols = ['a', 'b', 'c'] %}
select
    {% for col in cols %}
    nullif({{ col }}, "") as {{ col }}
    {% endfor %}
from tbl
```

But for crazy all-jinja models/macros, you lose a lot by not having any indentation. (e.g., the [incremental](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/include/global_project/macros/materializations/models/incremental/incremental.sql) materialization). 

Maybe the right answer is somewhere in between -- don't indent on the first open block, but do indent on any nested blocks?

Or we could allow this to be configurable by model, based on the presence of a `-- fmt: indent jinja` flag or similar?

**Comment by dave-connors-3** [comment]
hey @tconbeer! 

I think I lean towards your middle ground there most of the time! I lean towards indenting every new block of a Jinja operator (except consecutive `set` blocks -- it's only when compiled SQL formatting is _very important_ (i.e. like in the dbt_metrics package) that i'd think about leaving anything at the top level. That being said, if there's just one Jinja block, the SQL generally remains readable enough that the above example passes the smell test for me

**Comment by alrocar** [comment, marked as answer]
> But for crazy all-jinja models/macros, you lose a lot by not having any indentation. (e.g., the [incremental](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/include/global_project/macros/materializations/models/incremental/incremental.sql) materialization).

💯

I personally find it more natural everything indented à la python (and have received feedback in that sense as well)

I would expect the block in the top comment to be indented like this:

```sql
{% set cols = ['a', 'b', 'c'] %}
select
    {% for col in cols %}
        nullif({{ col }}, "") as {{ col }}
    {% endfor %}
from tbl
```

Would that make sense to you?
